spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/route/Route.java

```diff
@@ -140,6 +147,8 @@ public String toString() {
 
 		protected List<GatewayFilter> gatewayFilters = new ArrayList<>();
 
+		protected TimeoutConfig timeout = EMPTY_TIMEOUT_CONFIG;
```

Comment:
what is the advantage of having constant `EMPTY_TIMEOUT_CONFIG`? Why we just can't make it nullable?

Reply:
replaced with noArgs constructor
make it nullable will lead to a lot of extra null checks and made code ugly

Reply:
That's how `@ConfigurationProperties` work.